File: app/src/main/java/com/olehliskovych/placenote/di/global/WindowModule.java
package com.olehliskovych.placenote.di.global;

import android.view.Window;

import com.olehliskovych.placenote.di.scopes.ActivityScope;
import com.olehliskovych.placenote.ui.base.BaseActivity;

import dagger.Module;
import dagger.Provides;

@Module
public abstract class WindowModule {
    @ActivityScope
    @Provides
    static Window provideWindow(BaseActivity activity) {
        return activity.getWindow();
    }
}

File: app/src/main/java/com/olehliskovych/placenote/di/global/WindowModule.kt
package com.olehliskovych.placenote.di.global

import android.view.Window

import com.olehliskovych.placenote.di.scopes.ActivityScope
import com.olehliskovych.placenote.ui.base.BaseActivity

import dagger.Module
import dagger.Provides

@Module
object WindowModule {

    @JvmStatic
    @ActivityScope
    @Provides
    fun provideWindow(activity: BaseActivity): Window {
        return activity.window
    }
}
